src/js/helpers/searchIndexBuilder.js

```diff
@@ -25,25 +25,72 @@ const searchIndexBuilder = function(pages) {
       // - If so, Add it to our search index.
       janisPages.forEach( pageType => {
         if (page.node[pageType]) {
-          searchIndex.push({
+          const searchIndexData = {
             title: page.node[pageType].title,
             janisUrls: page.node.janisUrls,
-            pageType: pageType,
+            pageType: page.node.pageType,
             summary: page.node[pageType].mission || page.node[pageType].shortDescription || ""
-          })
+          }
+          if (pageType === "eventpage") {
+            searchIndexData.date = page.node[pageType].date
+            searchIndexData.startTime = page.node[pageType].startTime
+            searchIndexData.endTime = page.node[pageType].endTime
+            searchIndexData.location = page.node[pageType].locations[0]
+            searchIndexData.eventUrl = page.node.janisUrls[0]
+            searchIndexData.eventIsFree = page.node[pageType].eventIsFree
+            if (page.node[pageType].registrationUrl) {
+              searchIndexData.registrationUrl = page.node[pageType].registrationUrl
+            }
+            if (page.node[pageType].eventIsFree) {
+              searchIndexData.feesRange = ""
+            } else {
+              const feesRange = page.node[pageType].fees.edges.map(f=>"$"+f.node.fee)
+              searchIndexData.feesRange = feesRange.join('-')
+            }
+          }
+          if (pageType === "locationpage") {
+            searchIndexData.physicalStreet = page.node[pageType].physicalStreet
+            searchIndexData.physicalUnit = page.node[pageType].physicalUnit
+            searchIndexData.physicalCity = page.node[pageType].physicalCity
+            searchIndexData.physicalState = page.node[pageType].physicalState
+            searchIndexData.physicalZip = page.node[pageType].physicalZip
+          }
+          searchIndex.push(searchIndexData)
```

Comment:
Since this is happening during build, I've tried to make this as explicit as possible to keep the search index as lead as can be - which made this bit a bit 'chunky'.